### ISSUE TITLE
FMU - Memory Buildup Issue 

### ISSUE THREAD

**Issue by Primkins** [OpenModelica/OpenModelica#12225]
Created the FMU BuildingSystems.Applications.ClimateAnalyses.FreeFloatingTemperature using  OpenModelica v1.24.0-dev (64-bit) nightly build. 

Application crashes when running the FMU in Python interface with OmSimulator. Attached python script and the FMU. It shows the below error:

```
LOG_ASSERT        | debug   | C:/Users/DEMO/AppData/Local/Temp/OpenModelica/OMEdit/FFT/FFT.fmutmp/sources/gc/memory_pool.c:100: pool_expand: Assertion `0 && "Attempt to allocate an unusually large memory. The memory management does not seem to be working as intended. Please create an issue on https://github.com/OpenModelica/OpenModelica/issues."` failed.
error:   [fmi2logger] D:/TestFMU/test_tmp/test-q81hy4ke/temp/0001_A (logFmi2Call): fmi2CompletedIntegratorStep: terminated by an assertion.
```

 - OpenModelica Version: OpenModelica v1.24.0-dev (64-bit)
 - OMSimulator-v2.11Post 257.
- OS: Windows 11, 64 bit
 - BuildingSystems-2.0.0-beta
 
[Demo.zip](https://github.com/OpenModelica/OpenModelica/files/14945731/Demo.zip)

**Comment by Primkins** [comment]
@lochel  this is also FMU memory problem

**Comment by casella** [comment]
@Primkins this should hopefully be fixed by #15363. Can you please check with the next nightly build? Thanks!

**Comment by Impaler343** [comment]
Just tested this with the nightly buiild and nothing seems to have changed:

```
LOG_ASSERT        | debug   | /tmp/OpenModelica_pranavr/OMEdit/Finalf4ed/210.fmutmp/sources/gc/memory_pool.c:100: pool_expand: Assertion `0 && "Attempt to allocate an unusually large memory. The memory management does not seem to be working as intended. Please create an issue on https://github.com/OpenModelica/OpenModelica/issues."` failed.
getBestJumpBuffer got mmc_jumper=(nil), globalJumpBuffer=(nil)
Fatal Python error: Aborted
```

Could someone else confirm?

**Comment by Primkins** [comment]
> [@Primkins](https://github.com/Primkins) this should hopefully be fixed by [#15363](https://github.com/OpenModelica/OpenModelica/pull/15363). Can you please check with the next nightly build? Thanks! 

@casella Couldn't find the latest Nightly Built at this [address.](https://build.openmodelica.org/omc/builds/windows/nightly-builds/64bit/).  The last nightly built was April 18, 2026. Please provide the location of latest Nightly Builts.  




> Just tested this with the nightly buiild and nothing seems to have changed:
> 
> ```
> LOG_ASSERT        | debug   | /tmp/OpenModelica_pranavr/OMEdit/Finalf4ed/210.fmutmp/sources/gc/memory_pool.c:100: pool_expand: Assertion `0 && "Attempt to allocate an unusually large memory. The memory management does not seem to be working as intended. Please create an issue on https://github.com/OpenModelica/OpenModelica/issues."` failed.
> getBestJumpBuffer got mmc_jumper=(nil), globalJumpBuffer=(nil)
> Fatal Python error: Aborted
> ```
> 
> Could someone else confirm?


@Impaler343 May be you are working with April 18, 2026 nightly built. Please check.

**Comment by Impaler343** [comment]
Yes, I was pulling a stale nightly branch. Thanks for pointing it out

**Comment by adrpo** [comment]
We are having issues with the builds. I am working on fixing them.

**Comment by Impaler343** [comment]
I tried building from source and running and I still get the same error, might have built it wrong, but here it is:

```
# 1. Install all dependencies
sudo apt-get install -y git cmake make gcc g++ gfortran \
  liblapack-dev libblas-dev autoconf automake libtool \
  pkg-config openjdk-11-jdk python3 \
  libboost-all-dev \
  libcurl4-openssl-dev \
  uuid-dev \
  qt6-base-dev qt6-tools-dev qt6-tools-dev-tools libqt6svg6-dev \
  libqt6core5compat6-dev libxkbcommon-dev \
  libreadline-dev \
  qt6-webengine-dev \
  libopenscenegraph-dev

# 2. Clone with submodules
git clone --recurse-submodules https://github.com/OpenModelica/OpenModelica.git
cd OpenModelica

# 3. Configure
cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DOM_USE_CCACHE=OFF

# 4. Build (1-3 hours)
cmake --build build --parallel $(nproc)

# 5. Install
sudo cmake --install build
```

**Comment by casella** [comment]
@Primkins the Windows nightly is available now. Please try that and report.

**Comment by Primkins** [comment]
@arun3688 Please suggest the compatible version of the OMSimulator for the OpenModelica-v1.27.0-dev-390-g7f4d387df9-64bit.exe.

**Comment by arun3688** [comment]
@Primkins You can test with the shipped version of OMSimulator in nightly builds, if you are testing via pip then you should use `pip install OMSimulator==2.1.3`

**Comment by Primkins** [comment]
> I tried building from source and running and I still get the same error, might have built it wrong, but here it is:
> 
> ```
> # 1. Install all dependencies
> sudo apt-get install -y git cmake make gcc g++ gfortran \
>   liblapack-dev libblas-dev autoconf automake libtool \
>   pkg-config openjdk-11-jdk python3 \
>   libboost-all-dev \
>   libcurl4-openssl-dev \
>   uuid-dev \
>   qt6-base-dev qt6-tools-dev qt6-tools-dev-tools libqt6svg6-dev \
>   libqt6core5compat6-dev libxkbcommon-dev \
>   libreadline-dev \
>   qt6-webengine-dev \
>   libopenscenegraph-dev
> 
> # 2. Clone with submodules
> git clone --recurse-submodules https://github.com/OpenModelica/OpenModelica.git
> cd OpenModelica
> 
> # 3. Configure
> cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DOM_USE_CCACHE=OFF
> 
> # 4. Build (1-3 hours)
> cmake --build build --parallel $(nproc)
> 
> # 5. Install
> sudo cmake --install build
> ```

Please generate FMU using the latest nightly build.

**Comment by Primkins** [comment]
Installed the OMSimulator-2.1.3 using pip.
FMU attached in the stated issue, crashes at iteration 2337 due to memory buildup issue. 
FMU generated by latest nightly build (OpenModelica-v1.27.0-dev-390-g7f4d387df9-64bit.exe) resolves memory buildup issue in Python with OMSimulator. This item can be treated as closed.
